Adding initial SX1280 Driver APIs and abstraction

tommytt427 opened the pull request:
These changes are removing the old drivers and replacing them with a new API to be compliant with RTOS

tommytt427 committed:
feat(sx1280): initialize new FreeRTOS-compliant driver structure

Replaces the entire previous driver structure with a new structure.
designed to meet the requirements outlined in the Radio Firmware Project Spec:
- clean, high-level API
- thread-safe SPI wrapper using FreeRTOS critical sections.

tommytt427 committed:
feat(sx1280): initialize new FreeRTOS-compliant driver structure

tommytt427 committed:
SX1280 hardware abstraction

Refactoring sx1280 hardware level abstraction to be implemented

tommytt427 committed:
Add SX1280 command opcodes, expanded API

Defined SX1280 command opcodes in the header for clarity and expanded the public API in SX1280.h

jhammerberg commented:
Please keep the old driver in some form or file. You will probably use it as a reference.

tommytt427 committed:
Additions to interface, working on development of functions

This is a slight overhaul and development of the interface, as well as implementing important functions.
